src/components/RequireLogin.tsx

```diff
@@ -9,6 +8,7 @@ type Props = {
   tokenEndpoint: string;
   authenticatingComponent?: React.ReactNode; 
   children: React.ReactNode;
+  successfulAuthentication: (redirectAfter: string) => void;
```

Comment:
Lets call this `onSuccess`, and instead of `(redirectAfter: string)`, do `(state: string|null)`.

The `state` oauth2 parameter does not need to contain a redirect url. We use it as one, but it's really an implementation detail for us.

Reply:
The name of this variable should probably be renamed throughout, and not default to `/`